Added proper themes for GTK apps

SakibShahariar opened the pull request:
### Description

<!-- Provide a concise description of the changes made in this pull request. -->
This makes the gtk apps more vibrant with proper css for both gtk3 and gtk4 and probably flatpak too.

### Changes
- [x] Improved <!-- Optimized existing functionality -->
- [ ] Bug Fixes <!-- Fixes Scripts/Other -->
- [ ] Feature <!-- Added -->
- [ ] Documentation <!-- Changes related to the documentation -->
- [ ] Other <!-- Refactoring, cleanup, or non-functional changes -->

### Context

<!-- Why is this change necessary? What problem does it solve or what new feature does it add? -->
Regular theme doesn't look that good, that's why I want to add this.

### How Has This Been Tested?

<!-- Describe the steps you followed to test the changes. If applicable, mention the specific environment where the testing took place. -->
- [x] Tested on Arch Linux/Based Distro.
- [ ] Tested on Fedora Linux/Based Distro.
- [ ] Tested on openSuse.

### Checklist

Please ensure your pull request meets the following requirements:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes do not introduce new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes.

### Screenshots 

<!-- Add any screenshots to help explain your changes or to demonstrate the new feature. -->
<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/e79049ba-dece-4f09-8a07-5b3d2e7216ef" />
<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/378f5ec6-22c2-4e65-b128-35064c6a0e20" />


### Additional Notes

<!-- Add any other context, technical details, or considerations you'd like to share. -->
Check if gtk4 apps are looking right. Also check gtk3 apps, and flatpak apps. If doesn't work, we need to set theme using gsettings and for flatpak need to set up it's theme.

SakibShahariar committed:
updated gtk3 and gtk4 color path

SakibShahariar committed:
added different color file for gtk3

SakibShahariar committed:
Update gtk-colors.css

SakibShahariar committed:
added theme

mylinuxforwork commented:
Thank you vety much. I like what I see in your screenshot. But, Does This require  so many files? Is there a way to only change the colors without adding all the icons?

SakibShahariar commented:
I checked and they can be removed by editing the css files. But svg icons are really small and don't take much space(less than 1 mb total) . But if you want I can edit them.

SakibShahariar commented:
ok I will edit them. Btw, check if they work for different apps like normal gtk3 and gtk4, flatpak apps and root accessed  apps like gparted

mylinuxforwork commented:
> ok I will edit them. Btw, check if they work for different apps like normal gtk3 and gtk4, flatpak apps and root accessed apps like gparted

Great.

SakibShahariar committed:
Update gtk3-colors.css

SakibShahariar committed:
Update gtk3-colors.css

SakibShahariar committed:
Update gtk.css

SakibShahariar committed:
Delete dotfiles/.themes/Material-Gnome/gtk-4.0/assets directory

SakibShahariar committed:
Update gtk.css

SakibShahariar committed:
Delete dotfiles/.themes/Material-Gnome/gtk-3.0/assets directory

SakibShahariar committed:
Update colors.css

SakibShahariar committed:
Update gtk.css

SakibShahariar committed:
Update gtk.css

SakibShahariar committed:
Update colors.css

SakibShahariar committed:
Delete dotfiles/.themes/Material-Gnome/gtk-2.0 directory

SakibShahariar committed:
Delete dotfiles/.themes/Material-Gnome/gtk-4.0/thumbnail.png

SakibShahariar committed:
Delete dotfiles/.themes/Material-Gnome/gtk-3.0/thumbnail.png

SakibShahariar commented:
Done

SakibShahariar commented:
Ok.

SakibShahariar commented:
Ok.

CtorW commented:
When is the release, it's bothering me hahaha

SakibShahariar commented:
I guess he forgot

SakibShahariar committed:
Merge branch 'mylinuxforwork:main' into main

SakibShahariar committed:
Create merge-gtk.py for gtk apps to hot colors

Add a script to change colors live instead of relaunch or restart a app

SakibShahariar committed:
Rename merge-gtk.py to merge-gtk.py

fixed the folder name

SakibShahariar committed:
Add post_hook to gtk4 template configuration

mylinuxforwork commented:
Hi. I have seen that you have updated your PR again. Is it compatible with 2.12.2?

SakibShahariar commented:
The updated PR enables live GTK color updates, so when the theme changes, GTK apps instantly reflect the new colors without needing a restart.

It just adds a python script so there's no problem

SakibShahariar commented:
I think I should have created a different pr for this

SakibShahariar committed:
Delete dotfiles/.config/matugen/post-hook-scripts directory

SakibShahariar committed:
Remove post_hook from gtk4 templates config

Removed post_hook entry from gtk4 templates configuration.

SakibShahariar committed:
Merge branch 'mylinuxforwork:main' into main

SakibShahariar committed:
Add script to toggle GTK themes

This script toggles the GTK color scheme between dark and light modes based on the current setting.

SakibShahariar committed:
Add post_hook for GTK4 template processing

SakibShahariar committed:
Remove post_hook from gtk4 templates config

Removed post_hook entry from gtk4 templates configuration.

SakibShahariar committed:
Delete dotfiles/.config/matugen/post-hook-scripts directory
